### PR TITLE
implement crud products

### PR DESCRIPTION
**Pull request by Lanng21**
image

- edit: 
image

- delete:
image

### CODE REVIEW

File: src/pages/Home/index.tsx [hunk 1]
```diff
@@ -1,41 +1,160 @@
...
 function Home() {
+  const dispatch = useDispatch();
+  const products = useSelector((state: RootState) => state.products);
+
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const [products, setProducts] = useState<IProduct[]>([]);
+  const [selectedProduct, setSelectedProduct] = useState<IProduct | null>(null);
+
+  useEffect(() => {
+    async function fetchData() {
```
Comment (DuongPham-dev) [inline src/pages/Home/index.tsx]: es6

File: src/store/product/api.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { IProduct } from '../../types/Product';
+
+const API_BASE_URL = 'http://localhost:5000';
```
Comment (DuongPham-dev) [inline src/store/product/api.ts:4]: move to ENV

File: src/stylesheets/layouts/_header.scss [hunk 3]
```diff
@@ -4,11 +4,22 @@
   max-width: 1200px;
   margin: 0 auto;
   padding: 0 20px;
-  .header {
+  h1 {
```
Comment (DuongPham-dev) [inline src/stylesheets/layouts/_header.scss]: Shouldn't style on the tag name

File: src/stylesheets/layouts/_header.scss [hunk 4]
```diff
@@ -4,11 +4,22 @@
...
     margin-top: 50px;
     font-weight: 700;
     font-size: 60px;
     line-height: 81px;
     color: #0c0033;
```
Comment (DuongPham-dev) [inline src/stylesheets/layouts/_header.scss]: Color should be created variables then reusable

File: src/stylesheets/components/_form.scss [hunk 5]
```diff
@@ -6,6 +6,11 @@
       width: 100%;
       object-fit: cover;
     }
+    button {
+      border: none;
+      background: #fff;
```
Comment (DuongPham-dev) [inline src/stylesheets/components/_form.scss]: Color should be created variables then reusable

File: src/pages/Home/index.tsx [hunk 6]
```diff
@@ -1,41 +1,160 @@
...
+      key: 'price',
+      header: 'Price',
+      render: (product: IProduct) => product.price,
+    },
+    {
+      key: 'action',
+      header: 'Action',
+      render: (product: IProduct) => (
+        <div className="action-buttons">
+          <button type="button" onClick={() => handleEdit(product)}>
```
Comment (DuongPham-dev) [inline src/pages/Home/index.tsx:108]: should be created Component then reusable

File: src/pages/Home/index.tsx [hunk 7]
```diff
@@ -1,41 +1,160 @@
...
+  const handleDelete = async (productId: number) => {
+    try {
+      await deleteProductFromServer(productId);
+      dispatch(deleteProduct(productId));
+    } catch (error) {
+      throw new Error('Failed to delete product');
+    }
   };
 
+  const columns = [
```
Comment (DuongPham-dev) [inline src/pages/Home/index.tsx:72]: Please create a mocks folder and move this to it

File: src/pages/Home/index.tsx [hunk 8]
```diff
@@ -1,41 +1,160 @@
...
+    setIsModalOpen(true);
   };
 
-  const handleSubmit = (formData: IProduct) => {
-    const newProduct = {
-      ...formData,
-      id: Date.now(),
-    };
-    setProducts((prevProducts) => [...prevProducts, newProduct]);
+  const handleDelete = async (productId: number) => {
```
Comment (DuongPham-dev) [inline src/pages/Home/index.tsx]: use useCallback here

File: src/pages/Home/index.tsx [hunk 9]
```diff
@@ -1,41 +1,160 @@
...
+        await createProduct(newProduct);
+        dispatch(addProduct(newProduct));
+      } catch (error) {
+        throw new Error('Failed to create product');
+      }
+    }
+    closeModal();
+  };
+
+  const handleEdit = (product: IProduct) => {
```
Comment (DuongPham-dev) [inline src/pages/Home/index.tsx]: same case

File: src/components/Form/index.tsx [hunk 10]
```diff
@@ -1,19 +1,18 @@
...
 import Clear from '../../assets/images/x-close.svg';
 
 interface ModalFormProps {
   onClose: () => void;
   onSubmit: (formData: IProduct) => void;
+  selectedProduct: IProduct;
 }
 
-function ModalForm({ onClose, onSubmit }: ModalFormProps) {
+function ModalForm({ onClose, onSubmit, selectedProduct }: ModalFormProps) {
```
Comment (DuongPham-dev) [inline src/components/Form/index.tsx]: es6, please

File: src/main.tsx [hunk 11]
```diff
@@ -1,11 +1,15 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
+import { Provider } from 'react-redux';
```
Comment (DuongPham-dev) [inline src/main.tsx]: Why not use contextAPI but use redux?

File: db.json [hunk 12]
```diff
@@ -1,3 +1,24 @@
...
+      "quantity": 0,
+      "price": 0,
+      "status": "",
+      "type": "",
+      "brand": "",
+      "brandImage": "",
+      "id": 1693133955730
+    }
+  ]
+}
```
Comment (baonguyen26) [inline db.json:14]: github action

**Pull request closed without merging**